darwin/Platform.c

```diff
@@ -120,21 +120,65 @@ const MeterClass* const Platform_meterTypes[] = {
    NULL
 };
 
-static double Platform_nanosecondsPerMachTick = 1.0;
+static void Platform_getCPUBrandString(char *cpu_brand_string, size_t cpu_brand_string_size) {
+   if (sysctlbyname("machdep.cpu.brand_string", cpu_brand_string, &cpu_brand_string_size, NULL, 0) == -1) {
+      CRT_fatalError("Could not determine what CPU we are running on");
```

Comment:
Is this really fatal?  Only used if we're running under Rosetta2 conditions AFAICS.
And, yow, these workarounds are horrendously hacky.  :(

Reply:
No, it's not, this was just lazy error handling 😛 

This string should be available on all CPUs, and this function might be used elsewhere, so IDK what the [least astonishing](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) behaviour should be.

Perhaps setting `cpu_brand_string = '\0'`?

Reply:
Either / or - as mentioned, I'd rather we didn't leak this low-level information out of the WAR-test function.

Reply:
I warned a log message to `stderr` and set the brand string to `"UNKOWN!"`.

https://github.com/amomchilov/htop/blob/737dd931b7ed69bbf6b8a68986a18fb7c8a0ffd8/darwin/PlatformHelpers.c#L62-L66